app/controllers/api/subcollections/alert_actions.rb

```diff
@@ -0,0 +1,19 @@
+module Api
+  module Subcollections
+    module AlertActions
+      def alert_actions_query_resource(object)
+        MiqAlertStatusAction.where(:miq_alert_status_id => object.id)
+      end
+
+      def alert_actions_create_resource(object, type, _id, data)
+        data['miq_alert_status_id'] = object.id
```

Comment:
This is changing the data that's passed into this method - perhaps you could `dup` the data before you do this?